File: foldingshape/submissions/wrong_answer/kk_wrong_rotation.cpp
#include <algorithm>
#include <cmath>
#include <iostream>
using namespace std;
typedef long l;

int main() {
  int h, w, hh, ww;
  cin >> h >> w >> hh >> ww;
  int a = max(h, w), b = min(h, w), c = max(hh, ww), d = min(hh, ww);
  h = a;
  w = b;
  hh = c;
  ww = d;
  int count = 0;
  while (h > hh || w > ww) {
    if (h > hh) {
      h = ceil(h / 2.0);
      count++;
    }
    if (w > ww) {
      w = ceil(w / 2.0);
      count++;
    }
  }
  cout << count << endl;
  return 0;
}

File: foldingshape/submissions/wrong_answer/kk_wrong_rotation.py
import math


def solve():
    h, w = map(int, input().split())
    hh, ww = map(int, input().split())
    h, w = max(h, w), min(h, w)
    hh, ww = max(hh, ww), min(hh, ww)
    count = 0
    while h > hh or w > ww:
        if h > hh:
            h = math.ceil(h / 2)
            count += 1
        if w > ww:
            w = math.ceil(w / 2)
            count += 1
        h, w = max(h, w), min(h, w)
    return count


print(solve())
